### PR TITLE
Move 'push' funciton from pendingmessage to pendingmessages

### PR DESCRIPTION
**Pull request by xitronix**
Move push function to PendingMessages.

### CODE REVIEW

File: universal-login-relayer/lib/services/messages/PendingMessages.ts [hunk 1]
```diff
@@ -17,12 +18,25 @@ export default class PendingMessages {
...
     }
-    await this.signExecution(hash, message);
+    await this.addSignatureToPendingMessage(hash, message);
     return hash;
   }
 
-  private async signExecution(hash: string, message: SignedMessage) {
-    await this.executionsStore.get(hash).push(message);
+  private async addSignatureToPendingMessage(hash: string, message: SignedMessage) {
+    const pendingMessage = this.executionsStore.get(hash);
```
Comment (JustynaBroniszewska) [inline universal-login-relayer/lib/services/messages/PendingMessages.ts]: `executionStore` -> `pendingMessagesStore` ?

File: universal-login-relayer/lib/services/messages/PendingMessages.ts [hunk 2]
```diff
@@ -17,12 +18,25 @@ export default class PendingMessages {
...
+    }
+    if (pendingMessage.containSignature(message.signature)) {
+      throw new DuplicatedSignature();
+    }
+    const key = utils.verifyMessage(utils.arrayify(calculateMessageHash(message)), message.signature);
+    const keyPurpose = await pendingMessage.walletContract.getKeyPurpose(key);
+    if (keyPurpose.eq(0)) {
+      throw new InvalidSignature('Invalid key purpose');
+    }
+    pendingMessage.collectedSignatures.push({signature: message.signature, key});
```
Comment (JustynaBroniszewska) [inline universal-login-relayer/lib/services/messages/PendingMessages.ts]: Name it `addSignature` and move to pendingMessagesStore

File: universal-login-relayer/lib/services/messages/PendingMessages.ts [hunk 3]
```diff
@@ -17,12 +18,25 @@ export default class PendingMessages {
...
-    await this.executionsStore.get(hash).push(message);
+  private async addSignatureToPendingMessage(hash: string, message: SignedMessage) {
+    const pendingMessage = this.executionsStore.get(hash);
+    if (pendingMessage.transactionHash !== '0x0') {
+      throw new DuplicatedExecution();
+    }
+    if (pendingMessage.containSignature(message.signature)) {
+      throw new DuplicatedSignature();
+    }
+    const key = utils.verifyMessage(utils.arrayify(calculateMessageHash(message)), message.signature);
```
Comment (JustynaBroniszewska) [inline universal-login-relayer/lib/services/messages/PendingMessages.ts]: extract to `getKeyFromMessage`

File: universal-login-relayer/test/relayer/services/messages/PendingMessage.ts [hunk 4]
```diff
@@ -76,49 +73,12 @@ describe('Pending Execution', async () => {
...
-            await expect(pendingMessage.push(msg1))
-                .to.be.rejectedWith('Execution request already processed');
-        });
-    });
-
     describe('Concatenate', async () => {
         it('should concatenate two signatures', async () => {
-            await pendingMessage.push(msg0);
-            await pendingMessage.push(msg1);
+            const key0 = utils.verifyMessage(utils.arrayify(calculateMessageHash(msg0)), msg0.signature);
```
Comment (JustynaBroniszewska) [inline universal-login-relayer/test/relayer/services/messages/PendingMessage.ts]: extract

File: universal-login-relayer/lib/services/messages/PendingMessages.ts [hunk 5]
```diff
@@ -17,12 +18,25 @@ export default class PendingMessages {
...
+    const keyPurpose = await pendingMessage.walletContract.getKeyPurpose(key);
+    if (keyPurpose.eq(0)) {
```
Comment (JustynaBroniszewska) [inline universal-login-relayer/lib/services/messages/PendingMessages.ts]: add to commons `const INVALID_KEY = 0`

File: universal-login-relayer/lib/services/messages/PendingMessage.ts [hunk 6]
```diff
@@ -38,27 +36,12 @@ export default class PendingMessage {
...
-    this.collectedSignatures.push({signature: msg.signature, key});
-  }
-
   async isEnoughSignatures(requiredSignatures? : Number) {
     const requiredSignaturesCount = requiredSignatures || await this.walletContract.requiredSignatures();
     return this.collectedSignatures.length >= requiredSignaturesCount;
   }
 
-  async confirmExecution(transactionHash: string) {
+  confirmExecution(transactionHash: string) {
```
Comment (JustynaBroniszewska) [inline universal-login-relayer/lib/services/messages/PendingMessage.ts]: `confirmExecution` -> `confirmMessage`

File: universal-login-relayer/lib/services/messages/PendingMessages.ts [hunk 7]
```diff
@@ -33,7 +47,7 @@ export default class PendingMessages {
     return  { ...message, signature: this.executionsStore.get(hash).getConcatenatedSignatures()};
   }
 
-  async confirmExecution(messageHash: string, transactionHash: string) {
+  confirmExecution(messageHash: string, transactionHash: string) {
```
Comment (JustynaBroniszewska) [inline universal-login-relayer/lib/services/messages/PendingMessages.ts]: same `confirmMessage`

File: universal-login-relayer/test/relayer/services/messages/PendingMessagesStore.ts [hunk 8]
```diff
@@ -40,7 +40,7 @@ describe('UNIT: PendingMessagesStore', async () => {
 
   it('containSignature should return true if signarture already collected', async () => {
     pendingMessagesStore.add(messageHash, pendingMessage);
-    await pendingMessage.push(message);
+    await pendingMessage.collectedSignatures.push({signature: message.signature, key: '0x'});
```
Comment (JustynaBroniszewska) [inline universal-login-relayer/test/relayer/services/messages/PendingMessagesStore.ts]: PendingMessages should use PendingMessagesStore to add new signature. `pendingMessagesStore.addSignature(messageHash)`

File: universal-login-relayer/lib/services/messages/PendingMessages.ts (hunk 1 above)
Comment (xitronix) [reply universal-login-relayer/lib/services/messages/PendingMessages.ts]: ✔️

File: universal-login-relayer/lib/services/messages/PendingMessage.ts (hunk 6 above)
Comment (marekkirejczyk) [reply universal-login-relayer/lib/services/messages/PendingMessage.ts]: `setTransactionConfirmed()`

File: universal-login-relayer/lib/services/messages/PendingMessages.ts (hunk 2 above)
Comment (marekkirejczyk) [reply universal-login-relayer/lib/services/messages/PendingMessages.ts]: Extract `addSignature` and move to `PendingMessagesStore`.

Comment (xitronix) [reply universal-login-relayer/lib/services/messages/PendingMessages.ts]: ✔️

File: universal-login-relayer/lib/services/messages/PendingMessages.ts (hunk 3 above)
Comment (xitronix) [reply universal-login-relayer/lib/services/messages/PendingMessages.ts]: ✔️

File: universal-login-relayer/lib/services/messages/PendingMessages.ts (hunk 5 above)
Comment (xitronix) [reply universal-login-relayer/lib/services/messages/PendingMessages.ts]: ✔️

File: universal-login-relayer/lib/services/messages/PendingMessage.ts (hunk 6 above)
Comment (xitronix) [reply universal-login-relayer/lib/services/messages/PendingMessage.ts]: When extract confirmExecution

File: universal-login-relayer/lib/services/messages/PendingMessages.ts (hunk 7 above)
Comment (xitronix) [reply universal-login-relayer/lib/services/messages/PendingMessages.ts]: When extract confirmExecution

File: universal-login-relayer/test/relayer/services/messages/PendingMessagesStore.ts (hunk 8 above)
Comment (xitronix) [reply universal-login-relayer/test/relayer/services/messages/PendingMessagesStore.ts]: ✔️

File: universal-login-relayer/test/relayer/services/messages/PendingMessage.ts (hunk 4 above)
Comment (xitronix) [reply universal-login-relayer/test/relayer/services/messages/PendingMessage.ts]: ✔️

**Pull request merged**